workflow/sync/sync_manager.go

```diff
@@ -66,6 +66,77 @@ func (cm *Manager) CheckWorkflowExistence() {
 	}
 }
 
+func getUpgradedKey(wf *wfv1.Workflow, key string, level SyncLevelType) string {
+	if wfv1.CheckHolderKeyVersion(key) == wfv1.HoldingNameV1 {
+		if level == WorkflowLevel {
+			return getHolderKey(wf, "")
+		}
+		return getHolderKey(wf, key)
+	}
+	return key
+}
+
+type SyncLevelType int
+
+const (
+	WorkflowLevel SyncLevelType = 1
+	TemplateLevel SyncLevelType = 2
+	ErrorLevel    SyncLevelType = 3
+)
+
+// legacy keys can be of the form
```

Comment:
Suggest `HoldingNameV1` instead of `legacy`

Reply:
Done